src/brushes/eraser_brush.class.js

```diff
@@ -0,0 +1,272 @@
+       * @returns
+       */
+      onMouseDown: function (pointer, options) {
+        if (!this.canvas._isMainEvent(options.e)) {
+          return;
+        }
+        this._prepareForDrawing(pointer);
+
+        var _this = this;
+        this.canvas.clone(function (c) {
+          _this.renderInit(c);
+        });
+      },
+
+      /**
+       * Use a clone of the canvas to render the non-erasable objects on the main context
+       */
+      renderInit(_canvas) {
+        var canvas = this.canvas;
+        if (canvas.backgroundImage && !canvas.backgroundImage.erasable) {
+          canvas.backgroundImage.clone(function (_clone) {
+            _canvas.setBackgroundImage(_clone);
+          });
+        } else {
+          _canvas.setBackgroundImage(null);
+        }
+        if (canvas.backgroundColor && canvas.backgroundColor instanceof fabric.Object &&
+          !canvas.backgroundColor.erasable) {
+          canvas.setBackgroundColor.clone(function (_clone) {
+            _canvas.setBackgroundColor(_clone);
+          });
+        } else {
+          _canvas.setBackgroundColor(canvas.backgroundColor);
+        }
+        _canvas.renderCanvas(
+          canvas.getContext(),
+          canvas.getObjects().filter(function (obj) {
+            return !obj.erasable;
+          })
+        );
+        this._render();
+        _canvas.dispose();
+      },
+
+
+      /**
+       * Restore ctx after _finalizeAndAddPath is invoked
+       * @param {fabric.Point} pointer
+       * @param {fabric.IEvent} options
+       * @returns
+       */
+      onMouseUp: function (pointer, options) {
+        var retVal = this.callSuper("onMouseUp", pointer, options);
+        this.canvas.renderAll();
+        return retVal;
+      },
+
+      _render: function () {
+        this.canvas.renderCanvas(
+          this.canvas.contextTop,
+          this.canvas.getObjects()
+        );
+        this.callSuper("_render");
+      },
+
+      /**
+       * Creates fabric.Path object to add on canvas
+       * @param {String} pathData Path data
+       * @return {fabric.Path} Path to add on canvas
+       */
+      createPath: function (pathData) {
+        var path = this.callSuper("createPath", pathData);
+        path.set({
+          globalCompositeOperation: "destination-out",
+          selectable: false,
+          evented: false
+        });
+        return path;
+      },
+
+      /**
+       * Adds path to existing eraser paths on object
+       * @private
+       * @param {fabric.Object} obj
+       * @param {fabric.Path} path
+       */
+      _addPathToObjectEraser: function (obj, path) {
+        var transformMatrix = fabric.util.invertTransform(
+          obj.calcTransformMatrix()
+        );
+
+        var clipObject;
+        if (!obj.eraser) {
+          clipObject = new fabric.StrokeClipPath(obj);
+        } else {
+          clipObject = obj.clipPath;
+        }
+        clipObject.addPath(path, transformMatrix);
```

Comment:
```suggestion
        fabric.util.addTransformToObject(path, transformMatrix);
        clipObject.addPath(path);
```

> I need help here, struggling calculating the transform matrix for the eraser path so I simply use ctx.transform.
This isn't good practice - it should somehow be calculated and added to the path's transform matrix